packages/pds/src/api/app/bsky/actor/getSuggestions.ts

```diff
@@ -62,14 +74,37 @@ export default function (server: Server, ctx: AppContext) {
         },
       }))
 
-      const lastResult = suggestionsRes.at(-1)
       return {
         encoding: 'application/json',
         body: {
           actors,
-          cursor: lastResult?.createdAt,
+          cursor: keyset.packFromResult(suggestionsRes),
         },
       }
     },
   })
 }
+
+type CountDidResult = { orderCount: number; did: string }
+type CountDidLabeledResult = { primary: number; secondary: string }
+
+export class CountDidKeyset extends GenericKeyset<
+  CountDidResult,
+  CountDidLabeledResult
+> {
+  labelResult(result: CountDidResult): CountDidLabeledResult {
+    return { primary: result.orderCount, secondary: result.did }
+  }
+  labeledResultToCursor(labeled: CountDidLabeledResult) {
+    return {
+      primary: labeled.primary.toString(),
+      secondary: labeled.secondary,
+    }
+  }
+  cursorToLabeledResult(cursor: Cursor) {
+    return {
+      primary: parseInt(cursor.primary),
```

Comment:
It's annoying, but we might consider a check on whether `cursor.primary` parses to a number.  Otherwise most random cursor strings sent-up here will probably get turned into 500s rather than 400s.  In other places I have something like this:
```ts
    if (isNaN(parsed)) {
      throw new InvalidRequestError('Malformed cursor')
    }
```

Reply:
ah nice good call 👌